src/main/java/org/grpc/service/EmployeeGrpcImpl.java

```diff
@@ -0,0 +1,211 @@
+        Employee employee = employeeRepository.findByWorkingNumber(request.getWorkingNumber());
+        if(employee == null){
+            Status status = Status.newBuilder()
+                    .setCode(Code.NOT_FOUND_VALUE)
+                    .setMessage("An employee with this working number not found")
+                    .build();
+            responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+            return;
+        }
+
+        responseObserver.onNext(convertEmployeeToEmployeeDTO(employee, shiftRepository));
+        responseObserver.onCompleted();
+    }
+
+    private static EmployeeDTO convertEmployeeToEmployeeDTO(Employee employee, ShiftRepository shiftRepository) {
+        ArrayList<Shift> assignedShifts = shiftRepository.findAllByEmployeeId(employee.getId());
+        List<ShiftDTO> shiftDTOs = new ArrayList<>();
+
+        assignedShifts.forEach(shift -> shiftDTOs.add(convertShiftToShiftDTO(shift)));
+
+
+        EmployeeDTO.Builder builder = EmployeeDTO.newBuilder();
+        builder.setId(employee.getId());
+        builder.setFirstName(employee.getFirstName());
+        builder.setLastName(employee.getLastName());
+        builder.setWorkingNumber(employee.getWorkingNumber());
+        builder.setEmail(employee.getEmail());
+        builder.setPassword(employee.getPassword());
+        builder.setAssignedShifts(ShiftDTOList.newBuilder().addAllDtos(shiftDTOs).build());
+
+        return builder.build();
+    }
+
+    private static Employee convertEmployeeDTOToEmployee(EmployeeDTO employeeDTO, ShiftRepository shiftRepository) {
+        return new Employee(
+                employeeDTO.getId(),
+                employeeDTO.getFirstName(),
+                employeeDTO.getLastName(),
+                employeeDTO.getWorkingNumber(),
+                employeeDTO.getEmail(),
+                employeeDTO.getPassword()
+        );
+    }
+
+    private static ShiftDTO convertShiftToShiftDTO(Shift shift) {
+        ZoneId localTimeZone = ZoneId.systemDefault();
+        ShiftDTO.Builder builder = ShiftDTO.newBuilder();
+
+        builder.setId(shift.getId());
+        builder.setStartDateTime(shift.getStartDateTime().atZone(localTimeZone).toInstant().toEpochMilli());
+        builder.setEndDateTime(shift.getEndDateTime().atZone(localTimeZone).toInstant().toEpochMilli());
+        builder.setTypeOfShift(shift.getTypeOfShift());
+        builder.setShiftStatus(shift.getShiftStatus());
+        builder.setDescription(shift.getDescription());
+        builder.setLocation(shift.getLocation());
+
+        return builder.build();
+    }
+
+    private static LocalDateTime convertEpochMillisToLDT(long epochMillis) {
+        Instant instant = Instant.ofEpochMilli(epochMillis);
+        ZoneId localTimeZone = ZoneId.systemDefault();
+        return instant.atZone(localTimeZone).toLocalDateTime();
+    }
+}
```

Comment:
What exactly does it do?

Reply:
Instant instant = ... -> transforms the long with miliseconds since epoch to an instant moment in time
localTimeZone gets your system's timezone
instant.atZone() -> return that instant of time in that timezone